src/main/java/com/fasterxml/jackson/databind/CacheProvider.java

```diff
@@ -0,0 +1,14 @@
+package com.fasterxml.jackson.databind;
+
+import com.fasterxml.jackson.databind.util.LookupCache;
+
+/**
+ * Interface that defines API for Cache configurations.
+ * 
+ * @since 2.16
+ */
+public interface CacheProvider {
+    
+    LookupCache<JavaType, JsonDeserializer<Object>> provideDeserializerCache();
```

Comment:
Javadoc?

Reply:
Also: would it make sense to pass `DeserializationConfig`, to let cache check configuration related to deserialization aspects?
(only if config instance is easily accessible from caller, which should be the case)

Reply:
Sorry, I meant that `DeserializationConfig` would be passed to provider when constructing `LookupCache`; not to `DeserializerCache`.